components/tab-panel/README.md

```diff
@@ -0,0 +1,93 @@
+TabPanel
+=======
+
+TabPanel is a React component to render an ARIA-compliant TabPanel. It has two sections: a list of tabs, and the view to show when tabs are chosen. When the list of tabs gets focused, the active tab gets focus (the first tab if there isn't one already). Using the arrow keys navigate between tabs AND select the newly focused tab at the same time.
+
+TabPanel's children are rendered using Function as a Child Component. The function is the tabName as an argument.
+
+## Usage
+
+Renders a TabPanel with each tab representing a paragraph with its title.
+
+```jsx
+
+import { TabPanel } from '@wordpress/components';
+
+const chooseTab = ( tabName ) => {
+	console.log( 'Choosing tab', tabName );
+};
+
+function MyTabPanel() {
+	return (
+		<TabPanel className="my-tab-panel"
+			activeClass="active-tab"
+			tabs={ [
+				{
+					name: 'tab1',
+					title: 'Tab 1',
+					className: 'tab-one',
+					onSelect: this.chooseTab,
+				},
+				{
+					name: 'tab2',
+					title: 'Tab 2',
+					className: 'tab-two',
+					onSelect: this.chooseTab,
+				},
+			] }>
+			{
+				( tabName ) => {
+					return <p>${ tabName }</p>;
+				}
+			}
+		</TabPanel>
+	)
+}
+```
+
+## Props
+
+The component accepts the following props:
+
+### className
+
+The class to give to the outer container for the TabPanel
+
+- Type: `String`
+- Required: No
+- Default: ''
+
+### orientation
+
+The orientation of the tablist (`vertical` or `horizontal`)
+
+- Type: `String`
+- Required: No
+- Default: `horizontal`
+
+### tabs
+
+A list of tabs where each tab is defined by an object with the following fields:
+
+1. name: String. Defines the key for the tab
+2. title: String. Defines the translated text for the tab
+3. className: String. Defines the class to put on the tab.
+4. onSelect: function taking tabName. This is called when the tab is selected.
```

Comment:
Don't you think we could use a single global `onSelect` prop providing the selected tabName as an argument instead of using a `onSelect` per tab?

Reply:
Yep. I thought about it. Initially thought that it was easy to list the same function as the handler for each tab when declaring it (especially if you were mapping to create them), so left it like this. But I have no issue with enforcing it to be the same for each tab.

Reply:
I see this has been changed elsewhere to use a global `onSelect`. The current line in the doc can now be removed, correct?